meshtal_analysis.py

```diff
@@ -215,9 +217,17 @@ def add_mesh(mesh1, mesh2):
         new_mesh.x_bounds = mesh1.x_bounds
         new_mesh.y_bounds = mesh1.y_bounds
         new_mesh.z_bounds = mesh1.z_bounds
+        
+        new_mesh.x_mids = mesh1.x_mids
+        new_mesh.y_mids = mesh1.y_mids
+        new_mesh.z_mids = mesh1.z_mids
+        
         new_mesh.data['value'] = new_val 
         new_mesh.data['rel_err'] = new_err
         new_mesh.data['Energy'] = mesh1.data['Energy']
+        new_mesh.data['x'] = new_mesh.x_mids
+        new_mesh.data['y'] = new_mesh.y_mids

```

Comment:
this is not right.  the x_mids variables has a each mid point only once.  the data['x'] will have them many times in order to make a regular 3d grid of points.  this should be mesh1.data['x'].  same for y and z